Google.PowerShell/PubSub/GcpsTopic.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                string formattedTopicname = PrefixProjectToTopic(topicName, Project);
+                Topic topic = new Topic() { Name = formattedTopicname };
+                ProjectsResource.TopicsResource.CreateRequest request = Service.Projects.Topics.Create(topic, formattedTopicname);
+                try
+                {
+                    Topic response = request.Execute();
+                    WriteObject(response);
+                }
+                catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Conflict)
+                {
+                    ErrorRecord errorRecord = new ErrorRecord(
+                        new ArgumentException($"Cannot create topic '{topicName}' because it already exists."),
+                        "TopicAlreadyExists",
+                        ErrorCategory.ResourceExists,
+                        topicName);
+                    WriteError(errorRecord);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Retrieves Google Cloud PubSub topics.
+    /// </para>
+    /// <para type="description">
+    /// Retrieves one or more Gooogle Cloud PubSub topics.
+    /// If -Topic is not used, the cmdlet will return all the topics under the specified project
+    /// (default project if -Project is not used). Otherwise, the cmdlet will return a list of topics
+    /// matching the topic names specified in -Topic and will throw an error for any topic that cannot be found.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Get-GcpsTopic</code>
+    ///   <para>This command retrieves all topics in the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcpsTopic -Topic "topic1", "topic2" -Project "my-project"</code>
+    ///   <para>
+    ///   This command retrieves 2 topics ("topic1" and "topic2") in the project "my-project".
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/pubsub/docs/publisher#list-topics-in-your-project)">
+    /// [Listing a Topic]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GcpsTopic")]
+    public class GetGcpsTopic : GcpsCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to check for topics. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The names of the topics to be retrieved.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [Alias("Name")]
+        public string[] Topic { get; set; }
```

Comment:
Should this be "Topics"? Or is the pattern in PowerShell not to pluralize things? It just seems strange to specify multiple items for a parameter named singular "Topic".